test/TrueLayer.AcceptanceTests/PaymentTests.cs

```diff
@@ -48,6 +52,39 @@ public async Task Can_create_payment(CreatePaymentRequest paymentRequest)
             hppUri.ShouldNotBeNullOrWhiteSpace();
         }
 
+        [Fact]
+        public async Task Can_create_payment_with_auth_flow()
+        {
+            var sortCodeAccountNumber = new AccountIdentifier.SortCodeAccountNumber("567890", "12345678");
+            var providerSelection = new Provider.Preselected("mock-payments-gb-redirect", "faster_payments_service")
+            {
+                Remitter = new RemitterAccount("John Doe", sortCodeAccountNumber),
+            };
+            var paymentRequest = CreateTestPaymentRequest(
+                providerSelection,
+                sortCodeAccountNumber,
+                authorizationFlow: new StartAuthorizationFlowRequest(
+                    providerSelection, new SchemeSelection.Preselected(),
+                    new Redirect(new Uri("http://localhost:3000/callback")))
+            );
+
+            var response = await _fixture.Client.Payments.CreatePayment(
+                paymentRequest, idempotencyKey: Guid.NewGuid().ToString());
+
+            response.StatusCode.ShouldBe(HttpStatusCode.Created);
+            response.Data.IsT3.ShouldBeTrue();
+            CreatePaymentResponse.Authorizing authorizing = response.Data.AsT3;
```

Comment:
worth defining an `AsAuthorizing()` utility on the entity?

Reply:
surely we can add it but I would like to do it for every union in a follow-up Pr if you don't mind

Reply:
Sure!